akka-remote/src/main/scala/akka/remote/AddressUidExtension.scala

```diff
@@ -35,13 +33,11 @@ class AddressUidExtension(val system: ExtendedActorSystem) extends Extension {
 
   private def arteryEnabled = system.provider.asInstanceOf[RemoteActorRefProvider].remoteSettings.Artery.Enabled
 
-  val longAddressUid: Long = {
-    val tlr = ThreadLocalRandom.current
-    if (arteryEnabled) tlr.nextLong()
+  val longAddressUid: Long =
+    if (arteryEnabled) system.uid
     // with the old remoting we need to make toInt.toLong return the same number
     // to keep wire compatibility
-    else tlr.nextInt().toLong
-  }
+    else system.uid.toInt.toLong
```

Comment:
but now we may put a different value in MDC than what we actually use in classic remoting, ok maybe not worth fixing that given that classic remoting is deprecated and we don't "break" anything

Reply:
true, I figured we shouldn't let concerns specific to already-deprecated akka-remote behavior complicate new akka-actor code